### PR TITLE
[RISCV] Install sifive_vector.h to riscv-resource-headers

### PR DESCRIPTION
**Pull request by 4vtomat**

### CODE REVIEW

**Comment by eopXD** [comment]
Maybe test coverage?

**Comment by 4vtomat** [comment]
> Maybe test coverage with `-I %resource_dir`?

Which file should I place the test?

**Comment by eopXD** [comment]
> > Maybe test coverage with `-I %resource_dir`?
> 
> Which file should I place the test?

`clang/test/Headers/`

File: clang/test/Headers/riscv-sifive-header.c [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// RUN: %clang_cc1 -triple riscv64 -target-feature +v %s
+// REQUIRES: riscv-registered-target
+
+// expected-no-diagnostics
+
+#include <sifive_vector.h>
```
Comment (eopXD) [inline clang/test/Headers/riscv-sifive-header.c:6]: Does this test case help verify we have `sifive_vector.h` installed?

Comment (4vtomat) [reply clang/test/Headers/riscv-sifive-header.c:6]: Lit test is run before install, how can we check it's properly installed?

Comment (eopXD) [reply clang/test/Headers/riscv-sifive-header.c:6]: I think we can't :(

**Review by eopXD — APPROVED** [review]

File: clang/lib/Headers/CMakeLists.txt [hunk 2]
```diff
@@ -623,6 +623,11 @@ install(
   EXCLUDE_FROM_ALL
   COMPONENT riscv-resource-headers)
 
+install(
```
Comment (topperc) [inline clang/lib/Headers/CMakeLists.txt:626]: Aren't we also missing `sifive_files` from `add_header_target("riscv-resource-headers" "${riscv_files};${riscv_generated_files}")`?

Is there any reason to have `sifive_files`? Should we put sifive_vector.h in `riscv_files`?

Comment (4vtomat) [reply clang/lib/Headers/CMakeLists.txt:626]: Yes, we are missing from add_header_target("riscv-resource-headers" "${riscv_files};${riscv_generated_files}").
I think we can just put sifive_vector.h in riscv_files, I'll have another patch for it, thanks!

**Review by topperc — APPROVED** [review]
LGTM

**Pull request merged**